### PR TITLE
metrics/influxdb: update dependency

### PR DESCRIPTION
**Pull request by holiman**
This is an alternative to #26928 . Instead of upgrading the whole influx package, it switches to use the smaller `influxdata/influxdb1-client` package. 

For influxdb-v2, we already use the smaller client package, so this change removes the large dependency. 

There are some api-changes in the handlng of points batch, and this PR needs to be tested in prod.

### CODE REVIEW

**Comment by holiman** [comment]
I used this PR to run geth `--dev` mode for a couple of minutes, shooting the metrics to grafana. 

https://geth-bench.ethdevops.io/d/QC1Arp5Wk/single-geth?orgId=1&var-host=martin_laptop&var-percentile=50&from=1679392506844&to=1679392639080

No obvious errors

**Comment by holiman** [comment]
Swapping this in on `bench07` now

**Comment by holiman** [comment]
Everything LGTM (15 minutes with a different PR, and 15 minutes with this PR: https://geth-bench.ethdevops.io/d/Jpk-Be5Wk/dual-geth?var-exp=bench07&var-master=bench07&var-percentile=50&from=1679568700769&to=1679570615983  )

File: metrics/influxdb/influxdb.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ type reporter struct {
 	namespace string
 	tags      map[string]string
 
-	client *client.Client
+	client client.Client
 
 	cache map[string]int64
```
Comment (fjl) [inline metrics/influxdb/influxdb.go:26]: I think this field is not used anymore

Comment (holiman) [reply metrics/influxdb/influxdb.go:26]: What do you mean? It's used in both Ping and `send`
```
		case <-pingTicker.C:
			_, _, err := r.client.Ping(0)
```

```
	return r.client.Write(bps)
```

**Comment by holiman** [comment]
Closed in favour of https://github.com/ethereum/go-ethereum/pull/26963

**Pull request closed without merging**